validator/src/bootstrap.rs

```diff
@@ -555,16 +531,17 @@ mod without_incremental_snapshots {
                                 gossip.take().unwrap();
                             cluster_info.save_contact_info();
                             gossip_exit_flag.store(true, Ordering::Relaxed);
-                            let (maximum_full_snapshot_archives_to_retain, maximum_incremental_snapshot_archives_to_retain) = if let Some(snapshot_config) =
+                            let (maximum_full_snapshot_archives_to_retain, maximum_incremental_snapshot_archives_to_retain, incremental_snapshot_archives_dir) = if let Some(snapshot_config) =
                                 validator_config.snapshot_config.as_ref()
                             {
-                                (snapshot_config.maximum_full_snapshot_archives_to_retain, snapshot_config.maximum_incremental_snapshot_archives_to_retain)
+                                (snapshot_config.maximum_full_snapshot_archives_to_retain, snapshot_config.maximum_incremental_snapshot_archives_to_retain, snapshot_config.incremental_snapshot_archives_dir.as_path())
                             } else {
-                                (DEFAULT_MAX_FULL_SNAPSHOT_ARCHIVES_TO_RETAIN, DEFAULT_MAX_INCREMENTAL_SNAPSHOT_ARCHIVES_TO_RETAIN)
+                                (DEFAULT_MAX_FULL_SNAPSHOT_ARCHIVES_TO_RETAIN, DEFAULT_MAX_INCREMENTAL_SNAPSHOT_ARCHIVES_TO_RETAIN, full_snapshot_archives_dir)
                             };
                             let ret = download_snapshot_archive(
                                 &rpc_contact_info.rpc,
-                                snapshot_archives_dir,
+                                full_snapshot_archives_dir,
+                                incremental_snapshot_archives_dir,
```

Comment:
In truth, I have a plan to remove this mod completely through another PR. I was going to do it right after the approval of this PR.
Magic strings (even empty ones) look like a crutch to me. I would have preferred to do it via Option, but I drilled down in code and it doesn't look like a minor edit in this case. Especially since it's actually dead code that I'm about to remove.
Given this information, do you insist on these suggestion? 😊